services/priceService.ts

```diff
@@ -62,23 +61,121 @@ export async function upsertCurrentPricesForNetworkId (responseData: IResponseDa
   })
 }
 
-export async function syncCurrentPrices (): Promise<boolean> {
+export async function upsertCurrentPricesForNetworkId (responseData: IResponseData, networkId: number): Promise<void> {
+  await upsertPricesForNetworkId(responseData, networkId, 0)
+}
+
+function validateTicker (ticker: string): void {
+  if (!Object.values(TICKERS).includes(ticker)) {
+    throw new Error(RESPONSE_MESSAGES.INVALID_TICKER_400.message)
+  }
+}
+
+function validatePriceAPIUrlAndToken (): void {
   if (appInfo.priceAPIURL === '') {
     throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_URL_400.message)
   }
-  const todayString = moment().format(PRICE_API_DATE_FORMAT)
+  if (appInfo.priceAPIToken === '') {
+    throw new Error(RESPONSE_MESSAGES.MISSING_PRICE_API_TOKEN_400.message)
+  }
+}
+
+function getPriceByDayURL (day: moment.Moment, ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/pricebydate/${appInfo.priceAPIToken}/${ticker}+${day.format(PRICE_API_DATE_FORMAT)}`
+}
+
+function getAllPricesByTickerURL (ticker: string): string {
+  validatePriceAPIUrlAndToken()
+  validateTicker(ticker)
+  return `${appInfo.priceAPIURL}/dailyprices/${appInfo.priceAPIToken}/${ticker}`
+}
+
+export async function getPriceForDayTicker (day: moment.Moment, ticker: string, attempt: number = 1): Promise<IResponseData | null> {
```

Comment:
I used `ticker` here because the api only accepts the `ticker`, as an input and also it outputs the ticker, there is no `id` to be used in these methods. 

<img width="442" alt="image" src="https://user-images.githubusercontent.com/124014001/219732096-21931330-ba27-4c75-a968-9d587287acea.png">

Reply:
Alright, makes sense